### ISSUE TITLE
Change default of `DeserializationFeature.FAIL_ON_NULL_FOR_PRIMITIVES` to `true` for 3.0

### ISSUE THREAD

**Issue by cowtowncoder** [FasterXML/jackson-databind#4858]
### Describe your Issue

As of Jackson 2.x, `DeserializationFeature. FAIL_ON_NULL_FOR_PRIMITIVES` defaults to false, allowing coercion from JSON `null` into Java `int`, `long` and so on.
For some users -- and especially on "low null" languages like Kotlin -- such coercion is rarely expected, so suggestion was made to change the default.

No decision is yet made: discussion for helping decision is here:

https://github.com/FasterXML/jackson-future-ideas/discussions/79

this issue is placeholder until decision is made.

**Comment by yawkat** [comment]
I assume title should say "change to `true`"?

Sounds like a good change to me.

**Comment by cowtowncoder** [comment]
@yawkat Thanks, yes fixed that typo. And looks like proposal is passing (5 to 1 voting as of now).

**Comment by cowtowncoder** [comment]
As per https://github.com/FasterXML/jackson-future-ideas/discussions/79, proposal approved, will work on this.

**Comment by JooHyukKim** [comment]
Added #4890 as well.

**Comment by JooHyukKim** [comment]
Per-test modification needed after #4884 merged.

- WIP(PR-made) : [jackson-modules-base](https://github.com/FasterXML/jackson-modules-base/pull/267) 
- WIP(locally) : jackson-dataformats-binary
- jackson-dataformats-text
- jackson-dataformat-xml
- jackson-module-kotlin
- jackson-module-scala

**Comment by JooHyukKim** [comment]
@cowtowncoder Made fixes to most of modules. all failing tests seem to be reasonable (nothing buggy).

May I ask for help with Scala module @pjfanning ?

**Comment by cowtowncoder** [comment]
@JooHyukKim yes, @pjfanning maintains Scala module and should be able to help.
